contracts/abstract/PrizeSplit.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+abstract contract PrizeSplit is OwnableUpgradeable {
+  using SafeMathUpgradeable for uint256;
+
+  enum TokenType {
+    Ticket,
+    Sponsorship
+  }
+
+  struct MultipleWinnersPrizeSplit {
+      address target;
+      uint16 percentage;
+      TokenType token;
+  }
+
+  MultipleWinnersPrizeSplit[] internal _prizeSplits;
+
+  /**
+    * @dev Emitted when a new MultipleWinnersPrizeSplit config is added.
+  */
+  event PrizeSplitSet(address indexed target, uint16 percentage, TokenType token, uint256 index);
+
+  /**
+    * @dev Emitted when a MultipleWinnersPrizeSplit config is removed.
+  */
+  event PrizeSplitRemoved(uint256 indexed index);
+
+
+  /**
+    * @notice Award ticket to prize split target
+    * @dev Award ticket to prize split target using prize strategy function
+    * @param target Address of recipient.
+    * @param amount Amount to receive.
+  */
+  function _awardPrizeSplitTicketAmount(address target, uint256 amount) virtual internal;
+  
+  /**
+    * @notice Award sponsorship to prize split target
+    * @dev Award sponsorship to prize split target using prize strategy function
+    * @param target Address of recipient.
+    * @param amount Amount to receive.
+  */
+  function _awardPrizeSplitSponsorshipAmount(address target, uint256 amount) virtual internal;
+
+  /**
+    * @notice Set the prize split(s) configration.
+    * @dev Set the prize split(s) by passing an array of MultipleWinnersPrizeSplit structs.
+    * @param newPrizeSplits Array of MultipleWinnersPrizeSplit structs.
+  */
+  function setPrizeSplits(MultipleWinnersPrizeSplit[] memory newPrizeSplits) external onlyOwner {
```

Comment:
Can we use calldata here? `MultipleWinnersPrizeSplit[] calldata newPrizeSplits`does that save gas?

Reply:
Good catch.